pkg/operator/ceph/object/user/controller.go

```diff
@@ -152,6 +173,21 @@ func (c *ObjectStoreUserController) createUser(context *clusterd.Context, u *cep
 	}
 	objContext := object.NewContext(context, u.Spec.Store, u.Namespace)
 
+	// Wait with exponential backoff for the objectstore to be created
+	err := wait.ExponentialBackoff(backoff, func() (ok bool, err error) {
+		_, err = context.RookClientset.CephV1().CephObjectStores(u.Namespace).Get(u.Spec.Store, metav1.GetOptions{})
+		if err == nil {
```

Comment:
why not removing this condition and let the last return returning `true, nil`?

Reply:
I was thinking of 3 conditions,
1. When there is no error (return true and stop waiting immediately)
2. When there is an IsNotFound error (return false and wait)
3. When there is any other error (return true with error to exit out of wait and let admin handle the issue because this might not something that we expect to self heal)

Makes sense?

Reply:
The conditions make sense, i think @leseb is wondering if there is a simpler way to write those conditions. This seems like the simplest, although functionally the same as what you have:
```
if err != nil && errors.IsNotFound(err) {
  return false, nil
}
// err might be nil or non-nil
return true, err
```

Reply:
ack.